src/string.cr

```diff
@@ -2423,7 +2553,10 @@ class String
     self if !blank?
   end
 
-  def ==(other : self)
+  # Returns `true` if this string is equal to `*other*.
+  # Comparison is done byte-per-byte: if a byte is different from the corresponding
+  # byte, `false` is returned and so on.
```

Comment:
I don't think this really needs to be specified since there's only one valid encoding for the same string - whether comparing codepoints or bytes.

Reply:
Same formulation is already at <=> method. We need it here not to distinguish byte and char comparsion but to document how strings are compared at all - by real content and not for example by memory address or any other way.

Reply:
If it's similar to `<=>` then this is fine